### ISSUE TITLE
Local/custom viz

### ISSUE THREAD

**Issue by niconoe** [airbnb/caravel#1405]
Hello,

For a current project, I'd like to have the feature of adding custom/local visualizations to Caravel. That issue has been discussed in https://github.com/airbnb/caravel/issues/697, which is closed now.

I therefore plan to start exploring how I could make this improvement, but have a few questions:

- Is it still something the team is interested in with a good chance of eventually merging my branch (if works properly)?
- Do you know if someone else is already working on this?
- Any strong opinions about what I should and shouldn't do?

Thanks a lot!

**Comment by cons0l3** [comment]
I would love to have a seamless and effordless way of putting new vizualisations in or take unused out. Every visualisation could be encapsulated as a module/class/... with a defined structure (interface) that can be extended. it would require a vizualisation-manager to find all currently installed viz and make them renderable for the slices and dashboard...

**Comment by niconoe** [comment]
Thanks !

My idea is to make it work with "one directory per visualization". A first step would be to rework the existing stock viz like that, then in a second step to also load local stuff from a user directory. In that form, I don't think it would allow disabling existing viz like you suggest, but maybe we should go iteratively.

**Comment by cons0l3** [comment]
yes, splitting the viz into separate modules is the first step. abstracting the viz to conform to a common interface the second. a viz-manager or viz-factory can than be used through out the rest of caravel to get an instance of a named viz. a manager needs to know howto find a named viz and get a "instance" of it or delegate a call to it. definitely the last step in the journey...

**Comment by xrmx** [comment]
FYI disabling visualizations is already possible https://github.com/airbnb/caravel/blob/master/caravel/viz.py#L2031

**Comment by mistercrunch** [comment]
You can just override this in array in your caravel_config.py
https://github.com/airbnb/caravel/blob/master/caravel/config.py#L162

**Comment by niks-git** [comment]
@mistercrunch 
I tried creating a new viz type for "dual axis bar and line" graph by following your steps using the chord example, but despite everything, I am still not able to see my newly created viz type in the drop down list after i restart the superset server.

Note: In your example, the code maintains a list of viz_types, but the latest code on which I am working computes that list using globals().

Can you suggest where could I be going wrong?

Files added: dualaxis_bar_line.css/.jsx/.png
Files updated: main.js, viz.py, visTypes.js
Files not modified: package.json
